### PR TITLE
Update middleware.js

### PR DESCRIPTION
**Pull request by rishi-raj-jain**

### CODE REVIEW

**Comment by rishi-raj-jain** [comment]
@saimonkat can you help me in this PR? I want to make the homepage as static asset and not as an ISR page. Is there something we're fetching on the server that we need to revalidate on the homepage after it is deployed?

image

**Comment by saimonkat** [comment]
@rishi-raj-jain I guess the only one thing we need to revalidate on the homepage it's GitHub stars count. That's why we has to disable it in this PR: https://github.com/neondatabase/website/pull/2589 for error pages

But let me check, I'll research deeper soon

**Comment by rishi-raj-jain** [comment]
> @rishi-raj-jain I guess the only one thing we need to revalidate on the homepage it's GitHub stars count. That's why we has to disable it in this PR: #2589 for error pages
> 
> But let me check, I'll research deeper soon

I think we can avoid that atleast for / and /home since it'd fetch it during the initial render on the server.

**Comment by saimonkat** [comment]
@rishi-raj-jain ok as I see, you already removed revalidate from the main Homepage in this PR [#2621](https://github.com/neondatabase/website/pull/2621)

In this case we're free to remove it from other copies of Homepage, I updated in the last commit

**Comment by rishi-raj-jain** [comment]
@saimonkat somehow it still lands up being in ISR:
Screenshot 2024-12-12 at 12 54 42 AM

**Comment by saimonkat** [comment]
@rishi-raj-jain are you testing the current PR [Preview](https://neon-next-git-rishi-raj-jain-patch-1-neondatabase.vercel.app/)? Can you let me know how to test it please

**Comment by rishi-raj-jain** [comment]
> @rishi-raj-jain are you testing the current PR [Preview](https://neon-next-git-rishi-raj-jain-patch-1-neondatabase.vercel.app/)? Can you let me know how to test it please

Did you get a chance at my comment at https://github.com/neondatabase/website/pull/2624#issuecomment-2536925756?

**Comment by rishi-raj-jain** [comment]
I think this is rooting from wordpress.

**Comment by rishi-raj-jain** [comment]
In the next output it shows that it is a static page, but somehow the homepage it still marked as ISR function when deployed on Vercel.

**Comment by saimonkat** [comment]
@rishi-raj-jain it looks like the Vercel issue, please check
https://github.com/vercel/next.js/discussions/49451

> Vercel generates ISR functions for all pages in the app dir by default and occasionally tries to revalidate them at runtime

Anyway I think we good to merge this PR as soon as we don't need revalidate for home pages

**Comment by rishi-raj-jain** [comment]
Trying specifying no revalidate explicity via https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#revalidate.

**Comment by rishi-raj-jain** [comment]
Screenshot 2024-12-13 at 12 58 21 AM

**Comment by rishi-raj-jain** [comment]
image
vs
Screenshot 2024-12-13 at 1 34 47 AM

shared cache is now finally being used for longer times.

**Comment by rishi-raj-jain** [comment]
Screenshot 2024-12-13 at 1 59 40 AM

**Comment by saimonkat** [comment]
@rishi-raj-jain just noticed that we missed `/home` route for logged in users with `neon_login_indicator` to show our homepage somehow. Should we restore it?

**Comment by americano98** [comment]
Hi @rishi-raj-jain , could you please clarify how a client-side redirect could be faster than using middleware? For instance, if we add the `neon_login_indicator` cookie in your preview and reload the homepage, we first see the homepage render briefly before being redirected to the console. However, in the current middleware implementation in the `main` branch, there’s no initial page render; we’re redirected directly to the console if the cookie is present. Could you elaborate on the reasoning behind this approach?

**Comment by rishi-raj-jain** [comment]
> @rishi-raj-jain just noticed that we missed `/home` route for logged in users with `neon_login_indicator` to show our homepage somehow. Should we restore it?

I added a redirect from '/home' to '/' which takes care of it, does that work?

**Comment by rishi-raj-jain** [comment]
> Hi @rishi-raj-jain , could you please clarify how a client-side redirect could be faster than using middleware

It's faster for TTFB - now the state of the homepage is exactly as like a static asset, served asap.

>  we first see the homepage render briefly before being redirected to the console. 

Exactly what the original suggestion was - PPR but here we're defaulting to immediate script on the client instead of on the server.

**Comment by saimonkat** [comment]
@rishi-raj-jain 
> I added a redirect from '/home' to '/' which takes care of it, does that work?

Now it works, thanks! I see you rolled back the middlewares file with the correct conditions
As you understood, I meant the situation when when the user logged in and went to https://neon.tech/home to see the homepage

File: src/app/(home-page)/youtube/page.jsx [hunk 1]
```diff
@@ -33,4 +33,4 @@ const YoutubePage = () => (
 
 export default YoutubePage;
 
-export const revalidate = 60;
+export const revalidate = false;
```
Comment (amitdahan) [inline src/app/(home-page)/youtube/page.jsx:36]: Isn't `false` the default?

https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#revalidate

**Review by amitdahan — APPROVED** [review]

**Pull request merged**